src/awkward/operations/ak_unflatten.py

```diff
@@ -116,18 +121,22 @@ def _impl(array, counts, axis, highlevel, behavior):
         if counts.ndim != 1:
             raise ak._errors.wrap_error(ValueError("counts must be one-dimensional"))
 
-        if not issubclass(counts.dtype.type, np.integer):
+        if not np.issubdtype(counts.dtype, np.integer):
             raise ak._errors.wrap_error(ValueError("counts must be integers"))
 
-        current_offsets = backend.index_nplike.empty(len(counts) + 1, dtype=np.int64)
+        current_offsets = backend.index_nplike.empty(counts.size + 1, dtype=np.int64)
```

Comment:
So... `counts` was originally obtained via `to_layout` (layouts have `.length`, not `.size`), but then it was replaced by `to_backend_array`. I assume that means it could be NumPy, CuPy, a JAX DeviceArray, or a TypeTracerArray. All of these have a `.size`?

For a NumPy array, the `.size` is the product of `shape`, not its first element. Is there any way that the NumpyArray might be multidimensional? I suppose not, if it's a valid `counts`.

Reply:
Yes, a previous PR explicitly ensures that `counts` is a backend array, all of which have `.size`. We have an explicit check on `counts.ndim` to ensure that it's 1D!

Reply:
> I don't see any tests, though. (Tests for the specific case of ak.unflatten with typetracers.) Do they come up in dask-awkward?

Dask-awkward does test this, because it brought this to my attention from their test suite. However, we should really test this (I'm being lazy :face_with_head_bandage). I'll add some tests before we merge.